src/components/FilmList/helpers.tsx

```diff
@@ -0,0 +1,50 @@
+import { films } from "../../films";
+import { FilmCard } from "../FilmCard";
+import { Film } from "../../interfaces";
+
+const filterByGenre = (movies: Array<Film>, selectedGenre: string) => {
+  return movies.filter((el) => {
+    return el.genre.indexOf(selectedGenre) > -1;
+  });
+};
+
+const finedSerchedFilm = (movies: Array<Film>, searchFilm: string) => {
+  return movies.filter((el) => {
+    return el.name.toLowerCase().includes(searchFilm.toLowerCase());
+  });
+};
+
+const sorting = (movies: Array<Film>, sortBy: string) => {
+  return movies.sort((a: Film, b: Film) => {
+    if (sortBy === "name") {
+      if (a.name < b.name) {
+        return -1;
+      }
+      if (a.name > b.name) {
+        return 1;
+      }
+      return 0;
+    } else {
+      return a.year - b.year;
+    }
+  });
+};
+
+const renderFilms = (
+  selectedGenre: string,
+  searchFilm: string,
+  sortBy: string
```

Comment:
It is not a good practice.
try to decompose and divide this logic with different parent components. If your logic change or you need to add something new it will be hard to maintain this code. It is to many many dependencies here already, and you have not even put movies here as function argument. And imagine now, if you change (selectedGenre, searchFilm, sortBy) you always run 3 filter/sorting mechanisms.
My suggestion:
1) to have movies filtered by input string in your main App file. So you can import your mocked data there and just filter by user input. As a benefit you don't  need to put 'searchedFilm' prop to your 'Main', just 
`<Main movies={finedSerchedFilm(movies, searchFilm)} />`.
2) Then in your 'Main' you can do a filter by genre and sorting from the `movies`  prop, and put prepared movies into your `FilmList`, only one prop
`<FilmList movies={sorting(filterByGenre(movies, selectedGenre), sortBy)} />`
3) So now your FilmList just will render, without any logic.
Make sense?